sakura_core/outline/CFuncInfoArr.h

```diff
@@ -40,14 +40,14 @@ class CFuncInfo;
 //! アウトライン解析 データ配列
 class CFuncInfoArr {
 public:
-	CFuncInfoArr();	/* CFuncInfoArrクラス構築 */
-	~CFuncInfoArr();	/* CFuncInfoArrクラス消滅 */
-	CFuncInfo* GetAt( int nIdx );	/* 0<=の指定番号のデータを返す */
-	void AppendData( CFuncInfo* pcFuncInfo );	/* 配列の最後にデータを追加する */
+	CFuncInfoArr() = default;	/* CFuncInfoArrクラス構築 */
+	~CFuncInfoArr() = default;	/* CFuncInfoArrクラス消滅 */
+	CFuncInfo& GetAt(int nIdx) { return m_cFuncInfoArr[nIdx]; }
+	const CFuncInfo& GetAt(int nIdx) const { return m_cFuncInfoArr[nIdx]; }
 	void AppendData( CLogicInt nFuncLineCRLF, CLayoutInt nFuncLineLAYOUT, const WCHAR* pszFuncName,
 					 int nInfo, int nDepth = 0 );	/* 配列の最後にデータを追加する 2002.04.01 YAZAKI 深さ導入*/
 	void AppendData( CLogicInt nLogicLine, CLogicInt nLogicCol, CLayoutInt nLayoutLine, CLayoutInt nLayoutCol, const WCHAR*, const WCHAR*, int, int nDepth = 0 );	/* 配列の最後にデータを追加する 2010.03.01 syat 桁導入*/
-	int	GetNum( void ){	return m_nFuncInfoArrNum; }	/* 配列要素数を返す */
+	size_t GetNum( void ) const {	return m_cFuncInfoArr.size(); }	/* 配列要素数を返す */
```

Comment:
`GetNum` を呼び出している箇所にて芋づる式に int -> size_t 修正が必要になってしまうため、今回の対応では int のままが良いのではと思いました。

Reply:
確かに64bitビルドで警告が出るので戻り値の型を `size_t` に変えると呼び出し元を変更する必要が生じてしまいますね。
`GetAt` メソッドの引数は `int` のままなのと要素数も `int` で十分なはずなので元に戻しておきます。

Reply:
この int か size_t かついては https://github.com/sakura-editor/sakura/pull/1564#issuecomment-791940854 で話題になっていたのを思い出し、悩ましいところだと思いました。

Reply:
https://github.com/sakura-editor/sakura/pull/1679#issuecomment-846701414 でも言及されている UBSan(UndefinedBehaviorSanitizer) で問題が検出できるのかもしれないですがVC++ではまだ使えないし過信しない方が良さそうです。

https://github.com/sakura-editor/sakura/pull/1564#issuecomment-791940854  で k-kagariさんが

> Google C++ Style Guide は size_t を historical accident とまで呼んでいたりします。

と書かれていたので検索してみたところ https://google.github.io/styleguide/cppguide.html#Integer_Types の最後に該当する文が書かれていましたので引用します。

> On Unsigned Integers
>
> Unsigned integers are good for representing bitfields and modular arithmetic. Because of historical accident, the C++ standard also uses unsigned integers to represent the size of containers - many members of the standards body believe this to be a mistake, but it is effectively impossible to fix at this point. The fact that unsigned arithmetic doesn't model the behavior of a simple integer, but is instead defined by the standard to model modular arithmetic (wrapping around on overflow/underflow), means that a significant class of bugs cannot be diagnosed by the compiler. In other cases, the defined behavior impedes optimization.
>
> That said, mixing signedness of integer types is responsible for an equally large class of problems. The best advice we can provide: try to use iterators and containers rather than pointers and sizes, try not to mix signedness, and try to avoid unsigned types (except for representing bitfields or modular arithmetic). Do not use an unsigned type merely to assert that a variable is non-negative.

最後の文の `Do not use an unsigned type merely to assert that a variable is non-negative.` というのは自分が過去に犯した過ちなのできちんと覚えておこうと思いました。。

Reply:
> Google C++ Style Guide は size_t を historical accident とまで呼んでいたりします。

size_t は型名に `u` の文字が含まれないのもあってか、ぱっと見、符号なしには見えないのが危険ポイントな気がしますね。
うっかりラップアラウンドさせて暴走事故を起こさないよう気を付けたいと思います。
```cpp
// size_t GetNum()
for( size_t i = 0; i < GetNum() - 1; ++i ){
	...
```